Sources/SwiftDocC/Semantics/Metadata/Metadata.swift

```diff
@@ -132,6 +137,107 @@ public final class Metadata: Semantic, AutomaticDirectiveConvertible {
                 problems.append(Problem(diagnostic: diagnostic, possibleSolutions: [solution]))
             }
         }
+
+        var categorizedAvailability = [MetadataAvailability.Platform : [MetadataAvailability]]()
+        for availability in availability {
+            categorizedAvailability[availability.platform, default: []].append(availability)
+        }
+
+        for availabilityAttrs in categorizedAvailability.values {
+            guard availabilityAttrs.count > 1 else {
+                continue
+            }
+
+            let duplicateIntroduced = availabilityAttrs.filter({ $0.introduced != nil })
+            let duplicateBeta = availabilityAttrs.filter({ $0.isBeta })
+            let duplicateDeprecated = availabilityAttrs.filter({ $0.isDeprecated })
+            if duplicateIntroduced.count > 1 {
```

Comment:
I'm wondering if it's worth condensing these three checks into something like this? Would just reduce the amount of duplication below a bit.

```suggestion
            let duplicateIntroduced = (availabilityAttrs.filter({ $0.introduced != nil }), "introduced")
            let duplicateBeta = (availabilityAttrs.filter({ $0.isBeta }), "beta")
            let duplicateDeprecated = (availabilityAttrs.filter({ $0.isDeprecated }), "deprecated")
            let duplicates = [duplicatedIntroduced, duplicateBeta, duplicateDeprecated]
            for (duplicate, diagnosticLabel) in duplicates {
                guard duplicate.count > 1 else {
                    continue
                }
```


Could then use:

```swift
summary: "Duplicate \(MetadataAvailability.directiveName.singleQuoted) directive with 'diagnosticLabel' argument",
```

for the diagnostic.